dotCMS/src/main/java/com/dotcms/rest/api/v1/page/NavResource.java

```diff
@@ -116,7 +118,8 @@ public final Response loadJson(@Context final HttpServletRequest request, @Conte
                     response, Config.CONTEXT);
 
             final String path = (!uri.startsWith("/")) ? "/" + uri : uri;
-
+            //Force NavTool to behave as Live when rendering items
+            request.setAttribute(WebKeys.PAGE_MODE_PARAMETER, PageMode.LIVE);
```

Comment:
why do we need save the PageMode as a request attribute? we have it here

https://github.com/dotCMS/core/blob/d3a5a862e08c6111fa22b21b7ceba8e5e21e8a3b/dotCMS/src/main/java/com/dotmarketing/util/PageMode.java#L58

Reply:
now NavTool internally uses PageMode. And it works correctly when invoked from a page 
When invoked from a rest endpoint. it needs to resolve as LIVE. But it isn't therefore  I needed to override the attribute on the request so it resolves to LIVE. @freddyucv that's why

Reply:
done @freddyucv